platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ImmutableTransformExecution.java

```diff
@@ -57,17 +57,21 @@ public ImmutableTransformExecution(
     @Override
     public void visitIdentityInputs(InputVisitor visitor) {
         super.visitIdentityInputs(visitor);
-        // This is a performance hack. We could use the regular fingerprint of the input artifact, but that takes longer than
-        // capturing the normalized path and the snapshot of the raw contents, so we are using these to determine the identity
-        FileSystemLocationSnapshot inputArtifactSnapshot = fileSystemAccess.read(inputArtifact.getAbsolutePath());
-        visitor.visitInputProperty(INPUT_ARTIFACT_SNAPSHOT_PROPERTY_NAME, inputArtifactSnapshot::getHash);
+        visitor.visitInputFileProperty(INPUT_ARTIFACT_SNAPSHOT_PROPERTY_NAME, InputBehavior.NON_INCREMENTAL,
```

Comment:
```suggestion
        visitor.visitInputFileProperty(INPUT_ARTIFACT_PROPERTY_NAME, InputBehavior.INCREMENTAL,
```

I think we should do the same visiting as in `visitRegularInputs` here. Possibly extract this to a method, since we only do this as an identity input for immutable transform executions, not for the mutable ones.